libraries/preferences/impl/src/main/kotlin/io/element/android/libraries/preferences/impl/store/SessionPreferencesModule.kt

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2024 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.libraries.preferences.impl.store
+
+import com.squareup.anvil.annotations.ContributesTo
+import dagger.Module
+import dagger.Provides
+import io.element.android.features.preferences.api.store.SessionPreferencesStore
+import io.element.android.libraries.di.SessionScope
+import io.element.android.libraries.di.SingleIn
+import io.element.android.libraries.di.annotations.SessionCoroutineScope
+import io.element.android.libraries.matrix.api.user.CurrentSessionIdHolder
+import kotlinx.coroutines.CoroutineScope
+
+@Module
+@ContributesTo(SessionScope::class)
+object SessionPreferencesModule {
+    @Provides
+    @SingleIn(SessionScope::class)
+    fun providesSessionPreferencesStore(
+        defaultSessionPreferencesStoreFactory: DefaultSessionPreferencesStoreFactory,
+        currentSessionIdHolder: CurrentSessionIdHolder,
```

Comment:
Since this object has the scope `SessionScope`, can we inject directly the `sessionId: String` instead of the `CurrentSessionIdHolder`?

Oh actually `CurrentSessionIdHolder` is not what I though. It's not really a `Holder`, but more an accessor (or a provider). I think it will be simpler to provide the `sessionId` in `SessionMatrixModule` like some other fields and remove `CurrentSessionIdHolder`. Out of scope of this PR though.

Reply:
> I think it will be simpler to provide the sessionId in SessionMatrixModule like some other fields and remove CurrentSessionIdHolder. Out of scope of this PR though.

The problem is Anvil/Dagger doesn't understand that a `SessionId` is since it's Java based, for the DI library it's just a `String` and it won't allow you to inject it anywhere. That's why we had to create the holder wrapper. But maybe its name can be misleading.

Reply:
We had [annotations](https://github.com/element-hq/element-android/blob/main/matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/di/StringQualifiers.kt#L21) to be able to inject `String`s in the previous project and it was quite handy.